docs/ray.rst

```diff
@@ -0,0 +1,155 @@
+.. inclusion-marker-start-do-not-remove
+
+Horovod on Ray
+==============
+
+``horovod.ray`` allows users to leverage Horovod on `a Ray cluster <https://docs.ray.io/en/latest/cluster/index.html>`_.
+
+Currently, the Ray + Horovod integration provides a :ref:`RayExecutor API <horovod_ray_api>`.
+
+.. note:: The Ray + Horovod integration currently only supports a Gloo backend.
+
+Installation
+------------
+
+Use the extra ``[ray]`` option to install Ray along with Horovod.
+
+.. code-block:: bash
+
+    $ HOROVOD_WITH_GLOO=1 ... pip install 'horovod[ray]'
+
+See the Ray documentation for `advanced installation instructions <https://docs.ray.io/en/latest/installation.html>`_.
+
+
+Horovod Ray Job
+---------------
+
+The Horovod Ray integration offers a ``RayExecutor`` abstraction (:ref:`docs <horovod_ray_api>`),
+which is a wrapper over a group of `Ray actors (stateful processes) <https://docs.ray.io/en/latest/walkthrough.html#remote-classes-actors>`_.
+
+.. code-block:: python
+
+    from horovod.ray import RayExecutor
+
+    # Start the Ray cluster or attach to an exisint Ray cluster.
+    ray.init()
+
+    # Start num_hosts * num_slots actors on the cluster.
+    executor = RayExecutor(
+        setting, num_hosts=num_hosts, num_slots=num_slots, use_gpu=True)
+
+    # Launch the Ray actors on each machine.
+    # This will launch `num_slots` actors on each machine, each with
+    # 1 GPU allocated (set via CUDA VISIBLE DEVICES)
+    executor.start()
+
+
+All actors will be part of the Horovod ring, so ``RayExecutor`` invocations will be able to support arbitrary Horovod collective operations.
+
+Note that there is an implicit assumption on the cluster being homogenous in shape (i.e., all machines have the same number of slots available). This is simply
```

Comment:
Should be addressed following #2212.  We could consider allowing `hosts` to be a list of `host:slot` in a follow-up PR.

Reply:
Sounds good!